clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsRequest.java

```diff
@@ -59,16 +76,15 @@ protected Struct toStruct() {
     }
 
     @Override
-    public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
-        short versionId = version();
-        switch (versionId) {
+    public ApiVersionsResponse getErrorResponse(int throttleTimeMs, Throwable e) {
+        short version = version();
+        switch (version) {
             case 0:
                 return new ApiVersionsResponse(Errors.forException(e), Collections.<ApiVersionsResponse.ApiVersion>emptyList());
             case 1:
                 return new ApiVersionsResponse(throttleTimeMs, Errors.forException(e), Collections.<ApiVersionsResponse.ApiVersion>emptyList());
             default:
-                throw new IllegalArgumentException(String.format("Version %d is not valid. Valid versions for %s are 0 to %d",
-                        versionId, this.getClass().getSimpleName(), ApiKeys.API_VERSIONS.latestVersion()));
+                return new ApiVersionsResponse(Errors.UNSUPPORTED_VERSION, Collections.<ApiVersionsResponse.ApiVersion>emptyList());
```

Comment:
Should we document this in the javadoc? It doesn't follow the usual pattern of treating an unknown version as a bug. Also, do we need a test for this case in RequestResponseTest?

Reply:
Hmm.. Actually I don't think we need this change since unsupported versions are parsed as v0 with the `unsupportedRequestVersion` field indicating this case. That means that `version()` can be depended on to always return a valid version.